cmsl1t/plotting/base.py

```diff
@@ -24,14 +32,30 @@ def save_canvas(self, canvas=None, name=None):
         out_path = "{}/{}.{}".format(self.output_dir, name, self.output_format)
         canvas.SaveAs(out_path)
 
-    def build(self):
+    def build(self, *vargs, **kwargs):
+        """
+        We're reading in tree data so build the histograms etc from scratch.
+        Not used when we're restarting from previously generated histograms.
+        Calls the create_histograms() method which the derived class should implement.
+        """
+        self.create_histograms(*vargs, **kwargs)
+        self._is_built = True
+
+    def create_histograms(self):
         """
         Create the histograms that this plotter needs.
 
         This is not done in an init function so that we can by-pass this in the
         case where we reload things from disk
         """
-        raise NotImplementedError("build() needs to be implemented")
+        raise NotImplementedError("create_histograms() needs to be implemented")
+
+    def to_root(self, outfile):
+        """
+        Write histograms to a root file.
```

Comment:
No longer histograms being directly written, rather it's just this class, and ROOT takes care of the ROOT objects... update docstring